webrtc.html

```diff
@@ -11725,19 +11725,64 @@ <h3 id="mediastreamtrack-network-use">MediaStreamTrack</h3>
       <section>
         <h4>MediaTrackSupportedConstraints, MediaTrackCapabilities,
         MediaTrackConstraints and MediaTrackSettings</h4>
-        <p>The basics of <code>MediaTrackSupportedConstraints</code>,
-        <code>MediaTrackCapabilites</code>, <code>MediaTrackConstraints</code>
-        and <code>MediaTrackSettings</code> is outlined in [[!GETUSERMEDIA]].
-        However, the <code>MediaTrackSettings</code> for a
-        <code>MediaStreamTrack</code> sourced by an
-        <code><a>RTCPeerConnection</a></code> will only be populated with
-        members to the extent that data is supplied by means of the remote
-        <code><a>RTCSessionDescription</a></code> applied via
-        <code>setRemoteDescription</code> and the actual RTP data. This means
-        that certain members, such as <code>facingMode</code>,
-        <code>echoCancellation</code>, <code>latency</code>,
-        <code>deviceId</code> and <code>groupId</code>, will always be
-        missing.</p>
+        <p>The concept of constraints and constrainable properties, including
+        <code>MediaTrackConstraints</code>
+        (<code>MediaStreamTrack.getConstraints()</code>,
+        <code>MediaStreamTrack.applyConstraints()</code>), and
+        <code>MediaTrackSettings</code>
+        (<code>MediaStreamTrack.getSettings()</code>) are outlined in
+        [[!GETUSERMEDIA]]. However, the constrainable properties of tracks
+        sourced from a peer connection are different than those sourced by
+        <code>getUserMedia()</code>; the constraints and settings applicable to
+        <code><a>MediaStreamTrack</a></code>s sourced from a <a>remote
+        source</a> are defined here. The settings of a remote track represent
+        the latest frame received.
+        <code>MediaStreamTrack.getCapabilities()</code> MUST always return the
+        empty set and <code>MediaStreamTrack.applyConstraints()</code> MUST
```

Comment:
What about the case of empty constraints.
I believe WebKit implementation will return early in that case, which seems fine.
Note also https://github.com/w3c/mediacapture-main/pull/633.